python_modules/dagster/dagster/core/instance/__init__.py

```diff
@@ -52,7 +52,13 @@ def _dagster_home():
             'DAGSTER_HOME is not set, check is_dagster_home_set before invoking.'
         )
 
-    return os.path.expanduser(dagster_home_path)
+    dagster_home_path = os.path.expanduser(dagster_home_path)
+
+    if not os.path.isabs(dagster_home_path):
+        raise DagsterInvariantViolationError(
+            'DAGSTER_HOME must be absolute path'
```

Comment:
print out what we got here in the error

Reply:
Gotcha , seems like some tests failed too so I'll double check those too.

Reply:
I see a lot of the tests use `..` which is causing the test failures can't really find why dagster home gets set that way for other tests

Reply:
Actually original ticket was just about `.` but my code tests for both `.` and `..` so I'll remove latter from tests